sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BucketingFunction.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.dataflow.sdk.io;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Keep track of the minimum/maximum/sum of a set of timestamped long values.
+ * For efficiency, bucket values by their timestamp.
+ */
+public class BucketingFunction {
+  private static class Bucket {
+    private int numSamples;
+    private long combinedValue;
```

Comment:
See below - we already have a `SimpleFunction` in the SDK. But it may be that `BinaryCombineLongsFn` is an OK thing to re-use here (not that this requires reuse, necessarily). Your `combinedValue` will be a one-element `long[]` that you unpack in `get()`. Not sure if the optimizations ever come into play or not. But it does seem to be just the spec (associative, commutative)